pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -429,18 +462,39 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 				Eventually(k.UpdateStatus(mapiMachine, func() {
 					mapiMachine.Status.AuthoritativeAPI = mapiv1beta1.MachineAuthorityMachineAPI
 				})).Should(Succeed())
+
+				capiMachine = capiMachineBuilder.WithName(mapiMachine.Name).Build()
 			})
 
 			It("should successfully create the CAPI machine", func() {
-				Eventually(k.Get(
-					clusterv1resourcebuilder.Machine().WithName(mapiMachine.Name).WithNamespace(capiNamespace.Name).Build(),
-				)).Should(Succeed())
+				Eventually(k.Get(capiMachine)).Should(Succeed())
```

Comment:
With this change, you are updating the `capiMachine` object, does that matter? With the previous, it was a separate instance so it didn't overwrite the outer context?